compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -165,38 +312,91 @@ object Interactive {
                    && (includeReferences || isDefinition(tree))
                    && treePredicate(tree))
                 buf += SourceTree(tree, source)
+              traverseChildren(tree)
+            case tree: untpd.Inlined =>
+              traverse(tree.call)
             case _ =>
+              traverseChildren(tree)
           }
-          traverseChildren(tree)
         }
       }).traverse(topTree)
     }
 
     buf.toList
   }
 
-  /** Check if `tree` matches `sym`.
-   *  This is the case if `sym` is the symbol of `tree` or, if `includeOverriden`
-   *  is true, if `sym` is overriden by `tree`.
-   */
-  def matchSymbol(tree: Tree, sym: Symbol, includeOverriden: Boolean)(implicit ctx: Context): Boolean =
-    (sym == tree.symbol) || (includeOverriden && tree.symbol.allOverriddenSymbols.contains(sym))
-
-
   /** The reverse path to the node that closest encloses position `pos`,
    *  or `Nil` if no such path exists. If a non-empty path is returned it starts with
    *  the tree closest enclosing `pos` and ends with an element of `trees`.
    */
   def pathTo(trees: List[SourceTree], pos: SourcePosition)(implicit ctx: Context): List[Tree] =
     trees.find(_.pos.contains(pos)) match {
-      case Some(tree) =>
-        // FIXME: We shouldn't need a cast. Change NavigateAST.pathTo to return a List of Tree?
-        val path = NavigateAST.pathTo(pos.pos, tree.tree, skipZeroExtent = true).asInstanceOf[List[untpd.Tree]]
+      case Some(tree) => pathTo(tree.tree, pos.pos)
+      case None => Nil
+    }
 
-        path.dropWhile(!_.hasType).asInstanceOf[List[tpd.Tree]]
-      case None =>
-        Nil
+  def pathTo(tree: Tree, pos: Position)(implicit ctx: Context): List[Tree] =
+    if (tree.pos.contains(pos)) {
+      // FIXME: We shouldn't need a cast. Change NavigateAST.pathTo to return a List of Tree?
+      val path = NavigateAST.pathTo(pos, tree, skipZeroExtent = true).asInstanceOf[List[untpd.Tree]]
+      path.dropWhile(!_.hasType).asInstanceOf[List[tpd.Tree]]
     }
+    else Nil
+
+  def contextOfStat(stats: List[Tree], stat: Tree, exprOwner: Symbol, ctx: Context): Context = stats match {
+    case Nil =>
+      ctx
+    case first :: _ if first eq stat =>
```

Comment:
Why is the first statement handled specially?

Reply:
It's not. There's a tail recursion in the third clause.